[Serializer] Prevent that bad Ignore method annotations lead to incorrect results

astepin opened the pull request:
fix https://github.com/symfony/symfony/issues/45016

| Q             | A
| ------------- | ---
| Branch?       |  5.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #45016
| License       | MIT
| Doc PR        | na

By attaching the Ignore serializer annotation to a method that is not a set/get/has/is method, the property that was read before the method was marked as ignored. 
I have tweaked the behavior here so that it matches the other annotations. 

However, the change could now cause exceptions in older projects if they already have this bug built in. 
From my point of view, however, this is justifiable, because at the moment data may not be output correctly.

dunglas reviewed (approved):
LGTM

chalasr commented:
Thanks for the PR!
Can you check if 5.4 is also impacted? If yes, this PR should target 5.4.
Also, do other config formats suffer from the same issue? The behavior must be consistent across all formats so if it's the case the corresponding loaders should be patched as well.

astepin committed:
Prevent that bad Ignore method annotations lead to incorrect results

fix https://github.com/symfony/symfony/issues/45016

astepin commented:
Hey @chalasr, the bug is in fact already present in 5.4. I have now rebased to 5.4. 
Furthermore, I have modified the test so that it checks annotations and attributes, because the error behavior is identical.

chalasr commented:
@astepin Alright, by "other config formats" I was thinking of yaml and xml. After a quick look to these loaders it turns out they have no single validation rule regarding the accessor/mutator, so I'd say we can ignore them and go ahead here.
It seems like we do have an inconsistency on that point though, so feel free to have a look and submit a PR to add such validation if it's relevant and possible.

chalasr commented:
Thank you @astepin. And congratz on your first contrib!
